website/docs/commands/fmt.md

```diff
@@ -20,31 +20,45 @@ You can check formatting correctness using a `--check` flag:
 scala-cli fmt --check
 ```
 
-### Dialects
+### Scalafmt version and dialect
 
-Scala CLI also supports dialects that are passed to the formatter.
-This value is only used if there is no `.scalafmt.conf` file.
-However, if it exists, then all configuration should be placed there.
-For a list of all possible values, consult
-the [official Scala Dialects documentation](https://scalameta.org/scalafmt/docs/configuration.html#scala-dialects):
+Scala CLI `fmt` command supports passing the **dialect** and `scalafmt` **version** directly from the command line, using respectively the `--scalafmt-dialect` and `--scalafmt-version` options:
+```
+scala-cli fmt --scalafmt-dialect scala3 --scalafmt-version 3.5.8
+```
+You can skip passing either of those, which will make Scala CLI to infer a default value:
+- If a `.scalafmt.conf` file is present in the workspace and it has field defined, the value will be read from there, unless specified with Scala CLI options.
+- Otherwise, the default `scalafmt` **version** will be the latest one used by your Scala CLI version (so it is subject to change when updating Scala CLI) and the default **dialect** will be inferred based on the previously evaluated `scalafmt` version.
 
-```bash
-scala-cli fmt --dialect scala212
+#### Example 1
+
+``` text title=.scalafmt.conf
+version = "3.5.8"
+runner.dialect = scala212
 ```
 
-### Scalafmt version
+```bash
+scala-cli fmt --scalafmt-dialect scala213
+```
 
-At this time, Scala CLI reads a `scalafmt` version from `.scalafmt.conf` files. If the version is missing, Scala CLI
-throws an error, stating that users should declare an explicit Scalafmt version. Since Scalafmt `3.5.0`, this parameter
-is mandatory.
+For above setup `fmt` will use:
+- `version="3.5.8"` from the file
+- `dialect=scala213`, because passed `--scalafmt-dialect` option overrides dialect found in the file
 
-To configure the Scalafmt version, add the following to `.scalafmt.conf`. For example, to set the version to `3.5.0`,
-add the following line:
+#### Example 2
 
+``` text title=.scalafmt.conf
+version = "2.7.5"
 ```
-version = "3.5.0"
+
+```bash
+scala-cli fmt --scalafmt-version 3.5.8
 ```
 
+For above setup `fmt` will use:
+- `version="3.5.8"`, because passed `--scalafmt-version` option overrides version from the file
+- `dialect=scala3`, because dialect is neither passed as an option nor is it present in the configuration file, so it is inferred based on the previously evaluated version (`3.5.8`)
```

Comment:
uh, the dialect is inferred based on the Scala version, not the `scalafmt` version, no?
since `-S` wasn't passed, the default would be the newest release of dotty: `3.1.3`, as of writing this comment.

So it should be something more like:
```suggestion
- `dialect=scala3`, because dialect is neither passed as an option nor is it present in the configuration file, so it is inferred based on the Scala version; the Scala version wasn't explicitly specified in the command either, so it falls back to the default Scala version - the latest one, thus the resulting dialect is `scala3`. 
```

Reply:
Oh, thats right. My bad...